launch_ros/launch_ros/actions/node.py

```diff
@@ -370,6 +377,8 @@ def _perform_substitutions(self, context: LaunchContext) -> None:
         self.__final_node_name = ''
         if self.__expanded_node_namespace not in ['', '/']:
             self.__final_node_name += self.__expanded_node_namespace
+        elif self.__expanded_node_namespace == '':
+            self.__final_node_name += self.UNSPECIFIED_NODE_NAMESPACE
```

Comment:
@ivanpauno hmm, an empty string for a namespace is a valid argument for node creation (see [`rcl_node_init()`](https://github.com/ros2/rcl/blob/ab20449f19ea90670dc4419e9806e0c1b7b07d2d/rcl/include/rcl/node.h#L78-L79) documentation). 

I think that assuming an unspecified namespace is the root namespace would be the consistent thing to do.

Reply:
I didn't noticed it in #154, sorry about that.

Reply:
> @ivanpauno hmm, an empty string for a namespace is a valid argument for node creation (see rcl_node_init() documentation).

Sounds good.
I think that defaulting the namespace to `''` instead of `None` was a bad decision taken when `PushRosNamespace` action was added.
https://github.com/ros2/launch_ros/pull/153/commits/3933e356f12790e155968f0691bae43e066497b0 should fix the problem.

Reply:
Based on above's code (enforcing `self.__expanded_node_namespace` is a valid namespace or the `unspecified` token), wouldn't this code reduce to simply checking that `self.__expanded_node_namespace` isn't the root namespace before prepending?

Reply:
yeap, I've to add more tests to cover that case.

Reply:
Added more test cases in https://github.com/ros2/launch_ros/pull/153/commits/2e1f1ef3489547792941715983b9146ff24637dd